server/utils/authentication.js

```diff
@@ -10,17 +13,36 @@ export function getAllowedDomains(): string[] {
   return env ? env.split(",") : [];
 }
 
-export function signIn(
+export async function signIn(
   ctx: Context,
   user: User,
   team: Team,
   service: string,
-  isFirstSignin: boolean = false
+  isNewUser: boolean = false,
+  isNewTeam: boolean = false
 ) {
   if (user.isSuspended) {
     return ctx.redirect("/?notice=suspended");
   }
 
+  if (isNewTeam) {
+    // see: scenes/Login/index.js for where this cookie is written when
+    // viewing the /login or /create pages. It is a URI encoded JSON string.
+    const cookie = ctx.cookies.get("signupQueryParams");
+
+    if (cookie) {
+      try {
+        const signupQueryParams = pick(
+          JSON.parse(querystring.unescape(cookie)),
+          ["ref", "utm_content", "utm_medium", "utm_source", "utm_campaign"]
```

Comment:
Some filtering here to at least have control over which keys end up in this object